src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Publish/AvoidAssemblyLocationInSingleFile.cs

```diff
@@ -0,0 +1,158 @@
+                if (!string.Equals(isSingleFilePublish?.Trim(), "true", StringComparison.OrdinalIgnoreCase))
+                {
+                    return;
+                }
+                var includesAllContent = context.Options.GetMSBuildPropertyValue(
+                    MSBuildPropertyOptionNames.IncludeAllContentForSelfExtract, compilation, context.CancellationToken);
+                if (string.Equals(includesAllContent?.Trim(), "true", StringComparison.OrdinalIgnoreCase))
+                {
+                    return;
+                }
+
+                var properties = new List<IPropertySymbol>();
+                var methods = new List<IMethodSymbol>();
+
+                if (compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemReflectionAssembly, out var assemblyType))
+                {
+                    // properties
+                    addIfNotNull(properties, tryGetSingleSymbol<IPropertySymbol>(assemblyType.GetMembers("Location")));
+
+                    // methods
+                    methods.AddRange(assemblyType.GetMembers("GetFile").OfType<IMethodSymbol>());
+                    methods.AddRange(assemblyType.GetMembers("GetFiles").OfType<IMethodSymbol>());
+                }
+
+                if (compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemReflectionAssemblyName, out var assemblyNameType))
+                {
+                    addIfNotNull(properties, tryGetSingleSymbol<IPropertySymbol>(assemblyNameType.GetMembers("CodeBase")));
+                    addIfNotNull(properties, tryGetSingleSymbol<IPropertySymbol>(assemblyNameType.GetMembers("EscapedCodeBase")));
+                }
+
+                context.RegisterOperationAction(operationContext =>
+                {
+                    var access = (IPropertyReferenceOperation)operationContext.Operation;
+                    var property = access.Property;
+                    if (!contains(properties, property, SymbolEqualityComparer.Default))
+                    {
+                        return;
+                    }
+
+                    operationContext.ReportDiagnostic(access.CreateDiagnostic(LocationRule, property));
+                }, OperationKind.PropertyReference);
+
+                context.RegisterOperationAction(operationContext =>
+                {
+                    var invocation = (IInvocationOperation)operationContext.Operation;
+                    var targetMethod = invocation.TargetMethod;
+                    if (!contains(methods, targetMethod, SymbolEqualityComparer.Default))
+                    {
+                        return;
+                    }
+
+                    operationContext.ReportDiagnostic(invocation.CreateDiagnostic(GetFilesRule, targetMethod));
+                }, OperationKind.Invocation);
+
+                return;
+
+                static bool contains<T, TComp>(List<T> list, T elem, TComp comparer)
```

Comment:
We follow the Roslyn IDE conventions in this repo for local function names
```suggestion
                static bool Contains<T, TComp>(List<T> list, T elem, TComp comparer)
```

Reply:
yuck :)

Reply:
I'll do it, but I'd like to register my objection as the designer and author of local functions 😄